src/compounders/alien-base/AlienBaseCompounderSpot.sol

```diff
@@ -0,0 +1,85 @@
+ * Compounding can only be triggered if certain conditions are met and the initiator will get a small fee for the service provided.
+ * The compounding will collect the fees earned by a position and increase the liquidity of the position by those fees.
+ * Depending on current tick of the pool and the position range, fees will be deposited in appropriate ratio.
+ * @dev The contract prevents frontrunning/sandwiching by comparing the actual pool price with a pool price calculated from a TWAP.
+ */
+contract AlienBaseCompounderSpot is AlienBaseCompounder {
+    using FixedPointMathLib for uint256;
+
+    /* //////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    ////////////////////////////////////////////////////////////// */
+
+    /**
+     * @param initiatorShare The share of the fees paid to the initiator as reward, with 18 decimals precision.
+     * @param tolerance The maximum deviation of the actual pool price,
+     * relative to the price calculated with the TWAP, with 18 decimals precision.
+     * @dev The tolerance for the pool price will be converted to an upper and lower max sqrtPrice deviation,
+     * using the square root of the basis (one with 18 decimals precision) +- tolerance (18 decimals precision).
+     * The tolerance boundaries are symmetric around the price, but taking the square root will result in a different
+     * allowed deviation of the sqrtPriceX96 for the lower and upper boundaries.
+     */
+    constructor(uint256 compoundThreshold, uint256 initiatorShare, uint256 tolerance)
+        AlienBaseCompounder(0, initiatorShare, tolerance)
+    { }
+
+    /* ///////////////////////////////////////////////////////////////
+                    POSITION AND POOL VIEW FUNCTIONS
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Fetches all required position data from external contracts.
+     * @param id The id of the Liquidity Position.
+     * @return position Struct with the position data.
+     */
+    function getPositionState(uint256 id) public view override returns (PositionState memory position) {
+        // Get data of the Liquidity Position.
+        int24 tickLower;
+        int24 tickUpper;
+        (,, position.token0, position.token1, position.fee, tickLower, tickUpper,,,,,) =
+            AlienBaseLogic.POSITION_MANAGER.positions(id);
+        position.sqrtRatioLower = TickMath.getSqrtRatioAtTick(tickLower);
+        position.sqrtRatioUpper = TickMath.getSqrtRatioAtTick(tickUpper);
+
+        // Get data of the Liquidity Pool.
+        position.pool = AlienBaseLogic._computePoolAddress(position.token0, position.token1, position.fee);
+        (position.sqrtPriceX96,,,,,,) = IUniswapV3Pool(position.pool).slot0();
+
+        // Calculate the time weighted average tick over 300s.
+        // It is used only to ensure that the deposited Liquidity range and thus
```

Comment:
Comment does not entirely correct, twat prevent against flashloan attacks that manipultae pool for short amount of time to bring them out of balance.